examples/prometheus-operator/example.cue

```diff
@@ -0,0 +1,98 @@
+package prometheusOperator
+
+import "github.com/pollypkg/polly/schema:pollyschema"
+
+// NOTE we could make it an expectation that the emit value is
+// a pollyschema.PollyPackage
+prometheusOperator: pollyschema.PollyPackage & {
+	header: {
+		name: "prometheus-operator"
+		uri:  "github.com/pollypkg/polly/examples/prometheus-operator"
+		params: {}
+	}
+	signals: [
+		// List Errors
+		{
+			name: "ListErrors"
+			lang: "promql"
+			params: {job: string}
+			query: """
+                sum by (controller,namespace) (rate(prometheus_operator_list_operations_failed_total{job="\(params.job)"}[10m]))
+                /
+                sum by (controller,namespace) (rate(prometheus_operator_list_operations_total{job="\(params.job)"}[10m]))
+                > 0.4
+            """
+		},
+		// WatchErrors
+		{
+			name: "WatchErrors"
+			lang: "promql"
+			params: {job: string}
+			query: """
+              sum by (controller,namespace) (rate(prometheus_operator_watch_operations_failed_total{job="\(params.job)"}[10m]))
+              /
+              sum by (controller,namespace) (rate(prometheus_operator_watch_operations_total{job="\(params.job)"}[10m]))
+              > 0.4
+            """
+		},
+		// SyncFailed
+		{
+			name: "SyncFailed"
+			lang: "promql"
+			params: {job: string}
+			query: """
+              min_over_time(prometheus_operator_syncs{status=\"failed\",job="\(params.job)"}[5m]) > 0
+            """
+		},
+		// Reconcile Errors
+		{
+			name: "ReconcileErrors"
+			lang: "promql"
+			params: {job: string}
+			query: """
+                (sum by (controller,namespace) (rate(prometheus_operator_reconcile_errors_total{job="\(params.job)"}[5m])))
+                /
+                (sum by (controller,namespace) (rate(prometheus_operator_reconcile_operations_total{job="\(params.job)"}[5m]))) 
+                > 0.1
+            """
+		},
+		// NodeLookupErrors
+		{
+			name: "NodeLookupErrors"
+			lang: "promql"
+			params: {job: string}
+			query: """
+				rate(prometheus_operator_node_address_lookup_errors_total{job="\(params.job)"}[5m]) > 0.1
```

Comment:
One of the questions that's been weighing on me about signals is whether it's actually worth writing them for more straightforward queries - which this isn't SUPER straightforward, but close enough as an example to discuss. The general question here is, "when is it worth writing a signal?" It's interesting how this seems immediately _different_ from my intuitions about when it's worth exporting a function from a software library.

Like, if all we're doing is writing a signal for every metric, then that's just pointless laboriousness and why bother. I'm even kinda dubious about the utility of doing so with parameters, tbh, though that certainly helps. It seems like what really makes the parameters useful is when those parameters are part of some larger grouping - it's the relation, the shared-ness across data that is actually what we're after. That's also part of what datafaces are trying to capture. Things to figure out!

(No actual need for change here, just using it as an opportunity to raise a thing i've been thinking about)